src/main/java/BasketInformations.java

```diff
@@ -0,0 +1,98 @@
+package com.company;
+
+import java.util.HashMap;
+
+/**
+ * This class is a basket
+ */
+public class BasketInformations {
+
+	// The product of the basket
+	static HashMap<String, Integer> map = new HashMap<String, Integer>();
+
+	static String[] localProducts = new String[]{"123", "222", "44", "657"};
+
+	// The fact that the basket has promo code
+	private static boolean codeDePromotion = false;
+
+	public void addProductToBasket(String product, Integer price, boolean isPromoCode) {
+		if (isPromoCode) {
+			codeDePromotion = true;
+		} else {
+			map.put(product, price);
+		}
+	}
+
+	public Long getBasketPrice(boolean inCents) {
+		Integer v = 0;
+		for (String s : map.keySet()) {
+			v += map.get(s);
+		}
+		if (codeDePromotion) {
+			v -= 100;
+		}
+		return inCents ? v * 100 : Long.valueOf(v);
```

Comment:
12. If we changed the price datatype to BigDecimal (comment 6), we cannot return a Long if _inCents_ is false: the cents wil be lost.